libs/formats/src/zone/zone-validator.ts

```diff
@@ -0,0 +1,325 @@
+import { MapZones, Region, Track, Volume, Zone } from '@momentum/constants';
+import { Vec } from '@momentum/util-fn';
+
+// Contents of this file largely correspond to mom_zone_defs.cpp/h
+// Some of this implementation is kept deliberately quite C++ish so easiest to
+// keep both in sync.
+
+export const CURRENT_ZONE_FORMAT_VERSION = 1;
+
+// Size of a 4-byte (single precision) float, game code uses it as a default
+// value in some places.
+export const FLT_MAX = 3.40282347e38;
+
+export const MAX_COORD_FLOAT = 65536;
+export const MIN_COORD_FLOAT = -MAX_COORD_FLOAT;
+
+export const MAX_ZONE_REGION_POINTS = 64;
+
+export const MAX_TRACK_SEGMENTS = 256;
+export const MAX_SEGMENT_CHECKPOINTS = 256;
+export const MAX_ZONES_ALL_TRACKS = 512;
+
+export const MAX_STAGE_TRACKS = MAX_TRACK_SEGMENTS;
+export const MAX_BONUS_TRACKS = 256;
```

Comment:
I've changed these three `256` to `255` so I can fit the values in a single byte while 1-indexing (e.g. the first stage has `majorNum == 1`).